applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+
+#include <stdbool.h>
+#include <string.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+
+#include "cJSON.h"
+#include "cJSON_os.h"
+#include "cloud_codec.h"
+
+
+static const char * const sensor_type_str[] = {
+	[CLOUD_SENSOR_GPS] = "GPS",
+	[CLOUD_SENSOR_FLIP] = "FLIP",
+	[CLOUD_SENSOR_BUTTON] = "BUTTON",
+	[CLOUD_SENSOR_TEMP] = "TEMP",
+	[CLOUD_SENSOR_HUMID] = "HUMID",
+	[CLOUD_SENSOR_AIR_PRESS] = "AIR_PRESS",
+	[CLOUD_SENSOR_AIR_QUAL] = "AIR_QUAL",
+	[CLOUD_LTE_LINK_RSRP] = "RSRP",
+	[CLOUD_DEVICE_INFO] = "DEVICE",
+};
+
+/* --- A few wrappers for cJSON APIs --- */
+
+static int json_add_obj(cJSON *parent, const char *str, cJSON *item)
+{
+	cJSON_AddItemToObject(parent, str, item);
+
+	return 0;
+}
+
+static int json_add_str(cJSON *parent, const char *str, const char *item)
+{
+	cJSON *json_str;
+
+	json_str = cJSON_CreateString(item);
+	if (json_str == NULL) {
+		return -ENOMEM;
+	}
+
+	return json_add_obj(parent, str, json_str);
+}
+
+int cloud_encode_sensor_data(const struct cloud_sensor_data *sensor,
+				 struct cloud_data *output)
+{
+	int ret;
+
+	__ASSERT_NO_MSG(sensor != NULL);
+	__ASSERT_NO_MSG(sensor->data.buf != NULL);
+	__ASSERT_NO_MSG(sensor->data.len != 0);
+	__ASSERT_NO_MSG(output != NULL);
```

Comment:
This is a public API, so do you really want to `assert` if a parameter is invalid?

Reply:
I'll return error code instead